solana/programs/ntt-quoter/src/processor/request_relay.rs

```diff
@@ -0,0 +1,138 @@
+use anchor_lang::{
+    prelude::*,
+    system_program::{self, Transfer},
+};
+use example_native_token_transfers::queue::outbox::OutboxItem;
+use solana_program::{native_token::LAMPORTS_PER_SOL, sysvar};
+
+use crate::{
+    error::NttQuoterError,
+    state::{Instance, RegisteredChain, RelayRequest},
+    EVM_GAS_COST, WORMHOLE_TRANSCEIVER_INDEX,
+};
+
+#[derive(Accounts)]
+pub struct RequestRelay<'info> {
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    #[account(seeds = [Instance::SEED_PREFIX], bump = Instance::BUMP)]
+    pub instance: Account<'info, Instance>,
+
+    #[account(mut, address = instance.fee_recipient @ NttQuoterError::InvalidFeeRecipient)]
+    /// CHECK: leave britney alone (anchor complains about a missing check despite the address check)
+    pub fee_recipient: AccountInfo<'info>,
+
+    #[account(
+    seeds = [RegisteredChain::SEED_PREFIX, outbox_item.recipient_chain.id.to_be_bytes().as_ref()],
+    bump = registered_chain.bump,
+    constraint = registered_chain.base_price != u64::MAX @ NttQuoterError::RelayingToChainDisabled
```

Comment:
we could improve the readability here by adding an impl function `is_enabled` to `RegisteredChain` that checks if `base_price != u64::MAX`. Then this sentinel value would be closer to the struct definition (I'm assuming you didn't want to use `Option` to save a byte allocation)